lib/action_controller/serialization.rb

```diff
@@ -25,7 +25,8 @@ def get_serializer(resource, options = {})
           "Please pass 'adapter: false' or see ActiveSupport::SerializableResource.new"
         options[:adapter] = false
       end
-      serializable_resource = ActiveModel::SerializableResource.new(resource, options)
+      serializable_options = options.merge(controller: self)
```

Comment:
and do we need the whole controller? I'd rather not and looks like you just want the class name